src/model.jl

```diff
@@ -1,106 +1,275 @@
+  num_levels::T1
+  l_conv_chain::T2
+  l_down_chain::T3
+  r_up_chain::T4
+  r_conv_chain::T5
+  final_conv::T6
 end
 
-function Base.show(io::IO, u::Unet)
-  println(io, "UNet:")
+@functor Unet
+
+"""
+function Unet(;
+  in_channels = 1,
+  out_channels = 1,
+  num_fmaps = 64,
+  fmap_inc_factor = 2,
+  downsample_factors = [(2,2),(2,2),(2,2),(2,2)],
+  kernel_sizes_down = [[(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)]],
+  kernel_sizes_up = [[(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)]],
+  activation = NNlib.relu,
+  final_activation = NNlib.relu;
+  padding="same",
+  pooling_type="max"
+  )
+    creates a U-net model that can then be used to be trained and to perform predictions. A UNet consists of an initial layer to
+    create feature maps, controlled via `num_fmaps`. This is followed by downsampling and umsampling steps,
+    which obtain information from the downsampling side of the net via skip-connections, which are automatically inserted.
+    The down- and upsampling steps contain on each level a number of consequtive convolutions controlled via the arguments `kernel_sizes_down`
+    and `kernel_sizes_up` respectively.
+
+# Paramers
++ `in_channels`: channels of the input to the U-net
+
++ `out_channels`: channels of the output of the U-net
+
++ `num_fmaps`: number of feature maps that the input gets expanded to in the first step
 
-  for l in u.conv_down_blocks
-    println(io, "  ConvDown($(size(l[1].weight)[end-1]), $(size(l[1].weight)[end]))")
++ `fmap_inc_factor`: the factor that the feature maps get expanded by in every level of the U-net
+
++ `downsample_factors`: vector of downsampling factors of individual U-net levels
+
++ `kernel_sizes_down`: vector of vectors of tuples of individual kernel_sizes used in the convolutions on the way down
+  e.g. 5 lists of convolutions. 4 before downsampling and one final after the downsample, each with 2 consecutive 3x3 convolutions.
+
++ `kernel_sizes_up`: vector of vectors of tuples of individual kernel_sizes used in the convolutions on the way up (backwards)
+  similar but but after each upsampling step, starting from the top, but not initial one before upsampling.
+
++ `activation`: activation function after each convolution layer
+
++ `final_activation`: activation function for the final step
+
++ `padding="valid"`: method of padding during convolution and upsampling
+
++ `pooling_type="max"`: type of pooling
+
+# Example
+```jldoctest
+```
+"""
+function Unet(;  # all arguments are named and ahve defaults
+  in_channels = 1,
+  out_channels = 1,
+  num_fmaps = 64,
+  fmap_inc_factor = 2,
+  downsample_factors = [(2,2),(2,2),(2,2),(2,2)],
+  kernel_sizes_down = [[(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)]],
+  kernel_sizes_up = [[(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)], [(3,3), (3,3)]],
+  activation = NNlib.relu,
+  final_activation = NNlib.relu,
+  padding ="same",
+  pooling_type ="max"
```

Comment:
I would prefer to take in the function directly if possible - that way we simply call the input function and users can specify their own polling if preferable